src/nle.c

```diff
@@ -19,7 +19,7 @@
 #include <bzlib.h>
 #endif
 
-#define STACK_SIZE (1 << 15) /* 32KiB */
+#define STACK_SIZE (1 << 16) /* 64KiB */
```

Comment:
Perhaps a more fancy solution would be to do what pthreads do -- define a really large upper limit (8MiB or so) but use mmap to allocate the buffer and rely on page faults to grow actual memory usage. Perhaps @tscmoo can look into that :D

Reply:
I guess what could be done is to add a guard page at the end, a la https://man7.org/linux/man-pages/man3/pthread_attr_setguardsize.3.html

Reply:
The above was in fact not that wrong after all. All ways of getting buffers short of `mmap` may/do initialize the memory and therefore page everything in.

But it turns out fcontext does exactly the right thing anyway: [It`mprotect`s the bottom page of the new stack.](https://github.com/septag/deboost.context/blob/325fed7ee8b760e4cb79030a58dab74821fdcba9/source/stack.c#L137) I don't quite understand why this is a SIGBUS then, not a SIGSEGV.

However, I did learn: The page size of M1 MacBooks is 16384 and our 32KiB value was two pages, one of them protected and unavailable. Also, `create_fcontext_stack` actually picks a sensible default of called with `0`, which we should just do.

Reply:
Except that that default is currently broken due to a recent glibc change, reported at https://github.com/septag/deboost.context/issues/4.